src/System.IdentityModel.Tokens.Jwt/Validators.cs

```diff
@@ -169,7 +169,14 @@ public static void ValidateIssuerSecurityKey(SecurityKey securityKey, SecurityTo
             X509SecurityKey x509SecurityKey = securityKey as X509SecurityKey;
             if (x509SecurityKey != null)
             {
-                //validationParameters.CertificateValidator.Validate(x509SecurityKey.Certificate);
+                if (validationParameters.CertificateValidator == null)
+                    throw new SecurityTokenValidationException(ErrorMessages.IDX10232);
+
+                validationParameters.CertificateValidator.Validate(x509SecurityKey.Certificate);
+            }
+            else
```

Comment:
We need to check all key types. 
X509SecurityKey:
1. CertificateValidator 
2. IssuerSigningKeyValidator
Throw if both are null.

All other types.
use IssuerSigningKeyValidator, throw if null.